solana-labs/solana
# epoch reward vote backup

fork of the epoch_reward2 - update genesis to have authorize_vote_account to be different from vote_account. 
more close to real usage for vote account
node_key == authorized_key != vote_key.

- partitioned rewards
- cap reward interval by slot_in_epoch
- add partitioned reward enable flag
- fix bank reward tests
- add reward calc test
- fix test_compute_reward_calculation_signature
- add abiexample for crossbeam channel sender
- fix abi example tests
- update cargo.lock
- sort deps
- disable stake redelegate test
- fix calc request send
- don't add ancestor for calc bank
- Revert "don't add ancestor for calc bank"
- Revert "fix calc request send"
- send reward calc request in replay
- enable stake redelegation test
- remove cacl sender from blockstore
- send calc request from tpu banks
- remove log
- Revert "remove cacl sender from blockstore"
- add timings
- fix stake test
- fix stake_rewards_from_wrap test
- fix stake_merge_immediately_after_activation test
- another fix stake_merge_immediately_after_activation test
- disable test_two_unbalanced_stakes
- debug
- make calc/reward interval nonzero
- fix tvu reward calc send
- rm debug print
- panic when exceed max epoch calc result wait time
- remove todo
- purge old epoch reward result
- add get_reward_interval rpc method
- add rpc get_reward_interval_test
- fix rpc test
- add rpc get_reward_interval client test
- fix test
- clippy
- assert
- skip snapshot during reward interval
- fix snapshot test
- fix incremental snapshot tests
- fix snapshot with background svc
- revert
- return reward interval 1 for warmup epoch
- speedup incremental snapshot test
- fix rpc rward interval test
- comments out print in test
- typo
- fix bank reward interval test
- fix doc test
- fix comments
- fix test
- fix rpc client tests
- log total rewards number
- fix typo
- fix capitalization for partitioned rewards
- fix typo
- fix check
- refactor RewardsStoreMetric
- refactor reward cacl timing metrics
- tighten pub visibility
- clippy
- move pub to private
- add test for relinquish
- comments
- add reward account checks
- add locked reward count metrics
- add transaction error for locked reward account
- add in_reward param to load_accounts
- add locked reward account error to proto
- add lock account test
- fix bug vote instruction should be allowed during reward interval
- remove bad reward account test
- add reward_accounts_lock test
- assert instead of panic in test
- clippy unnecessary ref
- fix bug allow multiple vote instructions in one transaction in reward interval
- fix stake_account_lifetime tests. only access rewards accounts after reward interval
- add vote check tests
- comments
- fix convert::test::test_error_enums
- make diff command on new line in abi error message
- update bank delta slot abi digest
- typo
- fix stake redelegation test - update stake accounts after reward interval
- fix rpc tests
- fix stake_mege_immediately_after_activation test by adding test util to force end of reward interval
- fmt
- check vote_key and auth_voter_key must be different
- add feature set for vote key requirement
- add feature for enable partitioned epoch rewards
- assert auth_key and vote_key are different
- add authorized genesis creating fn
- update cluster info vote tests to use authorized votes
- fix clsuter info vote tests
- fix verify_vote in cluster_info_vote_listener
- fix cluster info vote tests
- fix authorized_vote key in vote account creation
- clean up debug print

#### Problem


#### Summary of Changes


Fixes #
